tiledb/sm/query/result_tile.cc

```diff
@@ -466,6 +466,67 @@ void ResultTile::compute_results_dense(
   }
 }
 
+/**
+ * A helper routine used in `compute_results_sparse<char>` to
+ * determine if a given string-valued coordinate intersects
+ * the given start and end range.
+ *
+ * @param c_offset The offset of the coordinate value
+ *    within `buff_str`.
+ * @param c_cize The size of the coordinate value.
+ * @param buff_str The buffer containing the coordinate value
+ *    at `c_offset`.
+ * @param range_start The starting range value.
+ * @param range_end The ending range value.
+ * @return uint8_t 0 for no intersection, 1 for instersection.
+ */
+inline uint8_t str_coord_intersects(
```

Comment:
Although I know it is used only in a single place, for consistency I would add its definition to `tiledb/sm/query/result_tile.h` (I do that pretty much everywhere else in the code).